src/SFA.DAS.Payments.EarningEvents.Application/Mapping/MappingExtensions.cs

```
@@ -1,23 +1,72 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 using ESFA.DC.ILR.FundingService.FM36.FundingOutput.Model.Abstract;
 using ESFA.DC.ILR.FundingService.FM36.FundingOutput.Model.Output;
 using FastMember;
 using SFA.DAS.Payments.Core;
 using SFA.DAS.Payments.Model.Core;
+using SFA.DAS.Payments.Model.Core.Entities;
 using PriceEpisode = ESFA.DC.ILR.FundingService.FM36.FundingOutput.Model.Output.PriceEpisode;
 
 namespace SFA.DAS.Payments.EarningEvents.Application.Mapping
 {
     public static class MappingExtensions
     {
         private static readonly TypeAccessor PeriodAccessor = TypeAccessor.Create(typeof(PeriodisedAttribute));
+        private static readonly TypeAccessor PeriodTextAccessor = TypeAccessor.Create(typeof(LearningDeliveryPeriodisedTextValues));
 
         public static decimal? GetPeriodValue(this PeriodisedAttribute periodisedValues, int period)
         {
             return (decimal?)PeriodAccessor[periodisedValues, "Period" + period];
         }
 
+        public static string GetPeriodTextValue(this LearningDeliveryPeriodisedTextValues periodisedValues, int period)
+        {
+            return PeriodTextAccessor[periodisedValues, "Period" + period].ToString();
+        }
+
+        public static ContractType[] GetContractTypesForLearningDeliveries(this List<LearningDelivery> learningDeliveries)
+        {
+            var periodisedTextValues = learningDeliveries.Where(x => x.LearningDeliveryPeriodisedTextValues != null)
+                .SelectMany(x =>
+                    x.LearningDeliveryPeriodisedTextValues.Where(l => l.AttributeName == "LearnDelContType"));
+
+            if (!periodisedTextValues.Any())
+            {
+                return new ContractType[0];
```

I think we should throw an exception here as we can't have payment with no contract type.